ipatests/test_integration/test_backup_and_restore.py

```diff
@@ -446,36 +449,133 @@ def test_full_backup_reinstall_restore_with_vault(self):
 
 
 class TestBackupAndRestoreWithReplica(IntegrationTest):
-    """Regression test for https://pagure.io/freeipa/issue/7234"""
-    num_replicas = 1
+    """
+        Regression tests for issues 7234 and 7455
+
+        https://pagure.io/freeipa/issue/7234
+            - check that oddjobd service is started after restore
+            - check new replica  setup after restore
+        https://pagure.io/freeipa/issue/7455
+            check that after restore and replication reinitialization
+                - users and CA data are at state before backup
+                - CA can be installed on existing replica
+                - new replica with CA can be setup
+    """
+    num_replicas = 2
     topology = "star"
 
     @classmethod
     def install(cls, mh):
+        cls.replica1 = cls.replicas[0]
+        cls.replica2 = cls.replicas[1]
         if cls.domain_level is None:
             domain_level = cls.master.config.domain_level
         else:
             domain_level = cls.domain_level
 
-        if cls.topology is None:
-            return
-        else:
-            tasks.install_topo(
-                cls.topology, cls.master, [],
-                cls.clients, domain_level
-            )
+        # Configure only master and one replica.
+        # Replica is configured without CA
+        tasks.install_topo(
+            cls.topology, cls.master, [cls.replica1],
+            cls.clients, domain_level,
+            setup_replica_cas=False
+        )
+
+    def create_csr(self, host):
+        request_path = host.run_command(['mktemp']).stdout_text
+        openssl_command = [
+            'openssl', 'req', '-new', '-nodes', '-out', request_path,
+            '-subj', '/CN=' + self.master.hostname
+        ]
+        host.run_command(openssl_command)
+        return request_path
+
+    def prepare_cert_signing_requests(self):
+        self.master.run_command(
+            ['ipa', 'service-add', 'TEST/' + self.master.hostname])
+        self.master_csr = self.create_csr(self.master)
+        self.replica1_csr = self.create_csr(self.replica1)
+        self.replica2_csr = self.create_csr(self.replica2)
+
+    def cleanup_cert_signing_requests(self):
+        self.master.run_command(['rm', self.master_csr])
+        self.replica1.run_command(['rm', self.replica1_csr])
+        self.replica2.run_command(['rm', self.replica2_csr])
+
+    def get_users(self, host):
+        res = host.run_command(['ipa', 'user-find'])
+        users = set()
+        for line in res.stdout_text.splitlines():
+            k, _unused, v = line.strip().partition(': ')
+            if k == 'User login':
+                users.add(v)
+        return users
+
+    def wait_replication_status(self, host, target_status):
```

Comment:
There is already a helper function ``wait_for_replication`` that does almost the same as this method. Would it be possible to enhance ``wait_for_replication`` rather than duplication features?

Reply:
extended tasks.wait_for_replication